doc/api/packages.md

```diff
@@ -819,10 +747,213 @@ conditional exports for consumers could be to add an export, e.g.
+within a particular `package.json` file’s package scope.
+
+Files ending with `.js` will be loaded as ES modules when the nearest parent
+`package.json` file contains a top-level field `"type"` with a value of
+`"module"`.
+
+The nearest parent `package.json` is defined as the first `package.json` found
+when searching in the current folder, that folder’s parent, and so on up
+until a node_modules folder or the volume root is reached.
+
+```json
+// package.json
+{
+  "type": "module"
+}
+```
+
+```console
+# In same folder as above package.json
+node my-app.js # Runs as ES module
+```
+
+If there is no package.json scope found, or if the nearest parent `package.json`
+lacks a `"type"` field, or contains `"type": "commonjs"`, or any other
+unsupported value, `.js` files are treated as CommonJS.
+
+```js
+// my-app.js, part of the same example as above
+import './startup.js'; // Loaded as ES module because of package.json
+```
+
+Regardless of the value of the `"type"` field, `.mjs` files are always treated
+as ES modules and `.cjs` files are always treated as CommonJS.
+
+### `"exports"`
+<!-- YAML
+added: v12.7.0
+changes:
+  - version:
+    - v12.16.0
+    - v13.2.0
+    pr-url: https://github.com/nodejs/node/pull/29978
+    description: Implement conditional exports.
+  - version:
+    - v12.15.0
+    - v13.7.0
+    pr-url: https://github.com/nodejs/node/pull/31001
+    description: Remove the `--experimental-conditional-exports` option.
+  - version:
+    - v12.16.0
+    - v13.7.0
+    pr-url: https://github.com/nodejs/node/pull/31008
+    description: Implement logical conditional exports ordering.
+-->
+
+```json
+{
+  "exports": "./index.js"
+}
+```
+
+* Type: {Object} | {string} | {string[]}
+
+The `"exports"` field allows defining the [entry points][] of a package when
+imported by name loaded either via a `node_modules` lookup or a
+[self-reference][] to its own name. It is supported in Node.js 12+ as an
+alternative to the [`"main"`][] that can support defining [subpath exports][]
+and [conditional exports][] while encapsulating internal unexported modules.
+
+[Conditional Exports][] can also be used within `"exports"` to define different
+package entry points per environment, including whether the package is
+referenced via `require` or via `import`.
+
+All paths defined in the `"exports"` must be relative file URLs starting with
+`./`.
+
+### `"main"`
+<!-- YAML
+added: v0.4.0
+-->
+
+```json
+{
+  "main": "./main.js"
+}
+```
+
+* Type: {string}
+
+The `"main"` field defines the script that is used when the [package directory
+is loaded via `require()`](modules.html#modules_folders_as_modules). Its value
+is interpreted as a path.
+
+```js
+require('./path/to/directory'); // This resolves to ./path/to/directory/main.js.
+```
+
+When a package has an [`"exports"`][] field, this will take precedence over the
+`"main"` field when importing the package by name.
+
+### `"imports"`
```

Comment:
Is this whole section copied from `esm.md` but not deleted from the old file? I don't think we want two copies of it.

Reply:
This was deleted from the top of this file, there is no imports section in esm.